### ISSUE TITLE
Add a mix-of-the-day section

### ISSUE THREAD

**Issue by yrew** [intergalacticfm/hotmixes.resty#19]
Add a section to the front page where we show a number of randomly selected mixes. 

"this way we'll be shining light on possibly forgotten filez"

This selection should be static as in for each day we want to display the same sets for every request.

**Comment by PanderMusubi** [comment]
If we choose one or two or three random mixes of the day, they should not be part of the recently added mixes.

Random selection will not be part of the RSS as RSS and podcast clients expect a growing thought trunkated list.

**Comment by yrew** [comment]
I'm going to think out loud. For this we require some sort of persistence. Currently the only system we have is the file system. If we want to select 3 mixes each day what solutions would apply?

- Run a cron job that generates a file with 3 mixes that will be used as input
- Write an algo that always selects the same three random mixes for a given day.
- Let a request to hotmixes trigger a generation of mixes
- Keep things in memory

I don;t know if those last two are even possible with the current stack but we'll find out

**Comment by PanderMusubi** [comment]
With the continuously growing number of mixes, already over 7000 or something, I think it is safe just to randomly choose three, or two or one every day. Change of the exact same selection is very slim. Not that we have to have a shuffle play for 11 tracks.

**Comment by PanderMusubi** [comment]
Something such as `find /blah/blah/ -type f -name '*.mp3'|sort -R|head -n 1` would be a good start.

**Comment by yrew** [comment]
Yes, something like that would work. We just need a way to store that result for a day otherwise it would repopulate with every request. Or at least that is my current understanding of how hotmixes works

**Comment by yrew** [comment]
Yes, I did some reading about Lua and how to handle this. Stuff like this is stored in files. Lua can write to them.

Every request we check a file, if its empty we populate it with

```
find /blah/blah/ -type f -name '*.mp3'|sort -R|head -n 1 would be a good start.
```

if it is populated and the last modified timestamp is less than 24h we read the links

if it is populated and the last modified timestamp is more than 24h we repopulate it.

**Comment by PanderMusubi** [comment]
![Screenshot from 2021-06-26 12-47-57](https://user-images.githubusercontent.com/198964/123510700-3e55d280-d67d-11eb-8ee2-2b431a1c7de1.png)

Would this be a way to show the mix of the day section?

**Comment by dromer** [comment]
> Yes, something like that would work. We just need a way to store that result for a day otherwise it would repopulate with every request. Or at least that is my current understanding of how hotmixes works

Return this on a separate endpoint and just let nginx cache this endpoint for 24hrs ?
